File: src/main/java/com/clearcont/clearcontapp/model/InputStreamMultipartFile.java
package com.clearcont.clearcontapp.model;

import com.amazonaws.util.IOUtils;
import org.jetbrains.annotations.NotNull;
import org.springframework.web.multipart.MultipartFile;

import java.io.*;

public record InputStreamMultipartFile(@NotNull InputStream inputStream, @NotNull String name) implements MultipartFile  {
    @Override
    public @NotNull String getName() {
        return name;
    }
    @Override
    public String getOriginalFilename() {
        return name;
    }
    @Override
    public String getContentType() {
        return "application/octet-stream";
    }
    @Override
    public boolean isEmpty() {
        return false;
    }
    @Override
    public long getSize() {
        try {
            return inputStream.available();
        } catch (IOException e) {
            return 0L;
        }
    }
    @Override
    public byte @NotNull [] getBytes() throws IOException {
        return IOUtils.toByteArray(inputStream);
    }
    @Override
    public @NotNull InputStream getInputStream() {
        return inputStream;
    }
    @Override
    public void transferTo(@NotNull File dest) throws IOException, IllegalStateException {
        try (OutputStream os = new FileOutputStream(dest)) {
            IOUtils.copy(inputStream, os);
        }
    }
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/models/InputStreamMultipartFile.kt
package br.com.clearcont.clearcontwebapp.models

import com.amazonaws.util.IOUtils
import org.springframework.web.multipart.MultipartFile
import java.io.File
import java.io.FileOutputStream
import java.io.IOException
import java.io.InputStream

class InputStreamMultipartFile() : MultipartFile {
    constructor(inputStream: InputStream, fileName: String) : this() {
    }

    override fun getName(): String {
        return name
    }

    override fun getOriginalFilename(): String {
        return name
    }

    override fun getContentType(): String {
        return "application/octet-stream"
    }

    override fun isEmpty(): Boolean {
        return false
    }

    override fun getSize(): Long {
        return try {
            inputStream.available().toLong()
        } catch (e: IOException) {
            0L
        }
    }

    @Throws(IOException::class)
    override fun getBytes(): ByteArray {
        return IOUtils.toByteArray(inputStream)
    }

    override fun getInputStream(): InputStream {
        return inputStream
    }

    @Throws(IOException::class, IllegalStateException::class)
    override fun transferTo(dest: File) {
        FileOutputStream(dest).use { os ->
            IOUtils.copy(inputStream, os)
        }
    }
}
